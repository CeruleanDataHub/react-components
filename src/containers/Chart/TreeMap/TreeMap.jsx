import React from "react";

import { Chart } from "../Chart";

const treeMapSeriesOptions = {
  type: "treemap",
  layoutAlgorithm: "squarified",
  alternateStartingDirection: true,
  levels: [
    {
      level: 1,
      layoutAlgorithm: "strip",
      dataLabels: {
        enabled: true,
        align: "left",
        verticalAlign: "top",
        style: {
          fontSize: "15px",
          fontWeight: "bold"
        }
      }
    }
  ]
};

export const TreeMap = ({ title, series }) => {
  const options = {
    title: {
      text: title
    },
    series: series.map(value => ({
      ...value,
      ...treeMapSeriesOptions
    }))
  };

  return <Chart title={title} series={series} options={options} />;
};

TreeMap.propTypes = {
  ...Chart.propTypes,
  /** Title of the chart */
  title: Chart.propTypes.title,
  /** Data to be graphed */
  series: Chart.propTypes.series
};

TreeMap.defaultProps = {
  title: Chart.defaultProps.title,
  series: Chart.defaultProps.series
};
