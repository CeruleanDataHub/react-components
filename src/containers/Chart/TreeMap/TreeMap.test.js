import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { TreeMap } from "./TreeMap";

const series = [
  {
    data: [
      {
        id: "A",
        name: "Apples",
        color: "#EC2500"
      },
      {
        id: "B",
        name: "Bananas",
        color: "#ECE100"
      },
      {
        id: "O",
        name: "Oranges",
        color: "#EC9800"
      },
      {
        name: "Anne",
        parent: "A",
        value: 5
      },
      {
        name: "Rick",
        parent: "A",
        value: 3
      },
      {
        name: "Peter",
        parent: "A",
        value: 4
      },
      {
        name: "Anne",
        parent: "B",
        value: 4
      },
      {
        name: "Rick",
        parent: "B",
        value: 10
      },
      {
        name: "Peter",
        parent: "B",
        value: 1
      },
      {
        name: "Anne",
        parent: "O",
        value: 1
      },
      {
        name: "Rick",
        parent: "O",
        value: 3
      },
      {
        name: "Peter",
        parent: "O",
        value: 3
      },
      {
        name: "Susanne",
        parent: "Kiwi",
        value: 2,
        color: "#9EDE00"
      }
    ]
  }
];

describe("Tree Map", () => {
  it("should render tree map", () => {
    const component = shallow(<TreeMap />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render tree map with data", () => {
    const component = shallow(<TreeMap series={series} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render tree map with title and data", () => {
    const component = shallow(<TreeMap title="Chart" series={series} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
