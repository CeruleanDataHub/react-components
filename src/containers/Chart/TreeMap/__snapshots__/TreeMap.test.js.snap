// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree Map should render tree map 1`] = `
<Chart
  containerProps={null}
  options={Object {}}
  series={Array []}
  title=""
  type="treemap"
  xAxis={null}
/>
`;

exports[`Tree Map should render tree map with data 1`] = `
<Chart
  containerProps={null}
  options={Object {}}
  series={
    Array [
      Object {
        "alternateStartingDirection": true,
        "data": Array [
          Object {
            "color": "#EC2500",
            "id": "A",
            "name": "Apples",
          },
          Object {
            "color": "#ECE100",
            "id": "B",
            "name": "Bananas",
          },
          Object {
            "color": "#EC9800",
            "id": "O",
            "name": "Oranges",
          },
          Object {
            "name": "Anne",
            "parent": "A",
            "value": 5,
          },
          Object {
            "name": "Rick",
            "parent": "A",
            "value": 3,
          },
          Object {
            "name": "Peter",
            "parent": "A",
            "value": 4,
          },
          Object {
            "name": "Anne",
            "parent": "B",
            "value": 4,
          },
          Object {
            "name": "Rick",
            "parent": "B",
            "value": 10,
          },
          Object {
            "name": "Peter",
            "parent": "B",
            "value": 1,
          },
          Object {
            "name": "Anne",
            "parent": "O",
            "value": 1,
          },
          Object {
            "name": "Rick",
            "parent": "O",
            "value": 3,
          },
          Object {
            "name": "Peter",
            "parent": "O",
            "value": 3,
          },
          Object {
            "color": "#9EDE00",
            "name": "Susanne",
            "parent": "Kiwi",
            "value": 2,
          },
        ],
        "layoutAlgorithm": "squarified",
        "levels": Array [
          Object {
            "dataLabels": Object {
              "align": "left",
              "enabled": true,
              "style": Object {
                "fontSize": "15px",
                "fontWeight": "bold",
              },
              "verticalAlign": "top",
            },
            "layoutAlgorithm": "strip",
            "level": 1,
          },
        ],
      },
    ]
  }
  title=""
  type="treemap"
  xAxis={null}
/>
`;

exports[`Tree Map should render tree map with title and data 1`] = `
<Chart
  containerProps={null}
  options={Object {}}
  series={
    Array [
      Object {
        "alternateStartingDirection": true,
        "data": Array [
          Object {
            "color": "#EC2500",
            "id": "A",
            "name": "Apples",
          },
          Object {
            "color": "#ECE100",
            "id": "B",
            "name": "Bananas",
          },
          Object {
            "color": "#EC9800",
            "id": "O",
            "name": "Oranges",
          },
          Object {
            "name": "Anne",
            "parent": "A",
            "value": 5,
          },
          Object {
            "name": "Rick",
            "parent": "A",
            "value": 3,
          },
          Object {
            "name": "Peter",
            "parent": "A",
            "value": 4,
          },
          Object {
            "name": "Anne",
            "parent": "B",
            "value": 4,
          },
          Object {
            "name": "Rick",
            "parent": "B",
            "value": 10,
          },
          Object {
            "name": "Peter",
            "parent": "B",
            "value": 1,
          },
          Object {
            "name": "Anne",
            "parent": "O",
            "value": 1,
          },
          Object {
            "name": "Rick",
            "parent": "O",
            "value": 3,
          },
          Object {
            "name": "Peter",
            "parent": "O",
            "value": 3,
          },
          Object {
            "color": "#9EDE00",
            "name": "Susanne",
            "parent": "Kiwi",
            "value": 2,
          },
        ],
        "layoutAlgorithm": "squarified",
        "levels": Array [
          Object {
            "dataLabels": Object {
              "align": "left",
              "enabled": true,
              "style": Object {
                "fontSize": "15px",
                "fontWeight": "bold",
              },
              "verticalAlign": "top",
            },
            "layoutAlgorithm": "strip",
            "level": 1,
          },
        ],
      },
    ]
  }
  title="Chart"
  type="treemap"
  xAxis={null}
/>
`;
